client/state/draft-feedback/reducer.js

```diff
@@ -0,0 +1,71 @@
+/**
+ * Internal dependencies
+ */
+import {
+	EDITOR_START,
+	EDITOR_STOP,
+	DRAFT_FEEDBACK_SHARE_ADD,
+	DRAFT_FEEDBACK_SHARE_REMOVE,
+	DRAFT_FEEDBACK_SHARE_REVOKE,
+	DRAFT_FEEDBACK_SHARE_RESTORE,
+	DRAFT_FEEDBACK_COMMENT_ADD,
+} from 'state/action-types';
+import { keyedReducer, withoutPersistence } from 'state/utils';
+
+const initialState = {};
+const initialDraftShareState = {
+	enabled: true,
+	comments: [],
+};
+
+const draftShareReducer = keyedReducer(
+	'emailAddress',
+	( state = initialDraftShareState, action ) => {
+		switch ( action.type ) {
+			case DRAFT_FEEDBACK_SHARE_ADD:
+				return initialDraftShareState;
+
+			case DRAFT_FEEDBACK_SHARE_REMOVE:
+				return undefined;
+
+			case DRAFT_FEEDBACK_SHARE_REVOKE:
+				return {
+					...state,
+					enabled: false,
+				};
+
+			case DRAFT_FEEDBACK_SHARE_RESTORE:
+				return {
+					...state,
+					enabled: true,
+				};
+
+			case DRAFT_FEEDBACK_COMMENT_ADD:
+				return {
+					...state,
+					comments: state.comments.concat( action.comment ),
+				};
+
+			default:
+				return state;
+		}
+	},
+);
```

Comment:
Nice work on this state stuff @brandonpayton. I would like to suggest however two bits here: use idiomatic Redux composition; don't create functions in the argument list of a function call (something I like to avoid in general unless for very simple arrow-functions)

We have `combineReducers()` which is a good fit for this state structure. I prefer to try and _never_ have state which is an object, instead relying on composition of reducers over simple data types instead.

The second bit is that with this use of `keyedReducer()` we can't test the inner reducer because they too coupled together. My habit is to create something like a `_Item` reducer and then fold over the `keyedReducer()`. This means a couple of things: first of all, we get a separate testable reducer, then second of all we can use the reducer outside of the wrapper.

```js
export const isEnabled = ( state = true, { type } ) => {
	switch ( type ) {
		case DRAFT_FEEDBACK_SHARE_REVOKE:
			return false;
		case DRAFT_FEEDBACK_SHARE_RESTORE:
			return true;
		default:
			return state;
	}
}

export const comments = ( state = [], { comments, type } ) =>
	type === DRAFT_FEEDBACK_COMMENTS_ADD
		? [ ...state, ...comments ]
		: state;

export const draftShareItem = combineReducers( {
	comments,
	isEnabled,
} );

export const draftShare = keyedReducer( 'emailAddress', draftShareItem );
```

here also I have tried to more clearly indicate the boolean type of the `enabled` since it could be just as easy to imagine `enabled` to mean a list of email addresses for which the share is enabled etc…